tests/test_states.py

```diff
@@ -220,6 +247,394 @@ def test_squeezed_coherent(self):
+
+        Args:
+            mu (array): means vector
+            cov (array): covariance matrix
+            func (function): function acting on the random variables X, P, XP,
+                returning a second order polynomial
+
+        Returns:
+            tuple: tuple of expectation value and variance.
+        """
+        X, P = np.mgrid[-7:7:.01, -7:7:.01]
+        grid = np.dstack((X, P))
+        XP = np.prod(grid, axis=2)
+
+        poly = func(X, P, XP)
+        PDF = multivariate_normal.pdf(grid, mu, cov)
+
+        Ex = simps(simps(poly * PDF, P[0]), X.T[0])
+        ExSq = simps(simps(poly**2 * PDF, P[0]), X.T[0])
+
+        var = ExSq - Ex**2
+
+        return Ex, var
+
+    def setUp(self):
+        """Set up."""
+        super().setUp()
+        self.eng, q = sf.Engine(self.num_subsystems, hbar=self.hbar)
+        self.eng.backend = self.backend
+        self.backend.reset(cutoff_dim=self.cutoff)
+
+        if isinstance(self.backend, backends.TFBackend):
+            raise unittest.SkipTest('The poly_quad_expectation method is not yet supported on the TF backend.')
+
+    def test_no_expectation(self):
+        """Test the case E(0), var(0)"""
+        self.logTestName()
+        state = self.circuit.state()
+
+        A = np.zeros([6, 6])
+        d = np.zeros([6])
+        k = 0
+
+        mean, var = state.poly_quad_expectation(A, d, k)
+        self.assertEqual(mean, 0)
+        self.assertEqual(var, 0)
+
+    def test_constant(self):
+        """Test the case E(k), var(k)"""
+        self.logTestName()
+        state = self.circuit.state()
+
+        A = np.zeros([6, 6])
+        d = np.zeros([6])
+        k = 0.543
+
+        mean, var = state.poly_quad_expectation(A, d, k)
+        self.assertEqual(mean, k)
+        self.assertEqual(var, 0)
+
+    def test_linear_vacuum(self):
+        """Test that the correct results are returned for the vacuum state."""
+        self.logTestName()
+
+        state = self.circuit.state()
+
+        A = np.zeros([6, 6])
+        # create an arbitrary linear combination
+        d = np.array([1, 1, 0, 1, 0, 0])
+        k = 0
+
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+        self.assertAlmostEqual(mean, 0, delta=self.tol)
+        self.assertAlmostEqual(var, len(d)*self.hbar/4, delta=self.tol)
+
+    def test_x_squeezed_coherent(self):
+        """Test that the correct E(x) is returned for the squeezed coherent state."""
+        self.logTestName()
+
+        A = None
+        d = np.array([1, 0, 0, 0, 0, 0])
+        k = 0
+
+        # prepare a squeezed displaced state
+        self.circuit.reset(pure=self.kwargs['pure'])
+        self.circuit.prepare_displaced_squeezed_state(a, r, phi, 0)
+
+        state = self.circuit.state()
+        mean, var = state.poly_quad_expectation(A, d, k, phi=self.qphi)
+
+        self.assertAlmostEqual(mean, self.mu[0], delta=self.tol)
+        self.assertAlmostEqual(var, self.cov[0, 0], delta=self.tol)
```

Comment:
Are these expected results independent of the `hbar` value?

Reply:
Yep, self.mu and self.cov are calculated taking hbar into account